demo/src/main/java/com/eis/geoCalendar/demo/Behaviour/EventMapBehaviour.java

```diff
@@ -0,0 +1,470 @@
+     *
+     * @param latLng The position on the map where the user clicked
+     */
+    @Override
+    public void onMapClick(LatLng latLng) {
+        if (bottomSheetBehaviour != null)
+            bottomSheetBehaviour.hide();
+
+        if (goToNavigatorView != null)
+            goToNavigatorView.setVisibility(INVISIBLE);
+
+        currentFocusMarker = null;
+    }
+
+    /**
+     * Called by a fragment of the application that returns the parameters to Create a new event and update the map
+     *
+     * @param pos         The position of the event to create
+     * @param description The description of the event to create
+     */
+    @Override
+    public void onEventReturn(LatLng pos, String description) {
+        GPSPosition eventPos = new GPSPosition(pos.latitude, pos.longitude);
+        Event event = new GenericEvent<>(eventPos, description);
+        if (onEventCreatedListeners != null && !onEventCreatedListeners.isEmpty())
+            for (OnEventCreatedListener listener : onEventCreatedListeners)
+                listener.onEventCreated(event);
+
+        Marker created = mMap.addMarker(new MarkerOptions().position(pos).title(description)); //automatically cuts title if too long
+        created.setTag(event);
+        moveMap(pos);
+    }
+
+    /**
+     * Called when user long presses an InfoWindow of Marker
+     *
+     * @param marker The marker whose InfoWindow was pressed
+     */
+    @Override
+    public void onInfoWindowLongClick(Marker marker) {
+        currentFocusMarker = marker;
+        callRemoveEventDialog(marker);
+    }
+
+    /**
+     * Opens a dialog that asks the user if the event has to be deleted
+     * If the user accepts, the dialog will call RemoveMark method of this instance
+     *
+     * @param marker The marker that has been selected for confirming deletion
+     */
+    protected void callRemoveEventDialog(Marker marker) {
+        if (removeEventDialog != null) {
+            removeEventDialog.setMarker(marker);
+            removeEventDialog.show(supportFragmentManager, REMOVE_EVENT_DIALOG_TAG);
+        }
+    }
+
+    /**
+     * Called by the RemoveEventDialog or the bottom sheet if the user confirms to delete the event
+     * Removes the Event mark from the map
+     * Makes forget the previously set currentFocusMarker
+     * Calls all the onEventRemovedListeners
+     *
+     * @param marker The marker to be removed
+     */
+    @Override
+    public void removeMark(Marker marker) {
+        currentFocusMarker = null;
+
+        if (onEventRemovedListeners != null && !onEventRemovedListeners.isEmpty())
+            for (OnEventRemovedListener listener : onEventRemovedListeners)
+                listener.onEventRemoved((Event) marker.getTag());
+
+        if (bottomSheetBehaviour != null)
+            bottomSheetBehaviour.hide();
+        if (goToNavigatorView != null)
+            goToNavigatorView.setVisibility(INVISIBLE);
+        marker.remove();

```

Comment:
In the specifications you declare 3 actions (line 333, 334 and 335) which are performed in a different chronological order within the method.
It is not important but for me, but since I am picky, I would have distributed these "actions" in the correct order.
I repeat, it is not important, you can RESOLVE this comment and nothing happened.

Reply:
Well, the specifications are there to avoid reading the code itself, so in this case, the order doesn't really matter, but for the sake of precision it could be done😂😂

Reply:
You are right, the specifications don't need to be integrated with code but have to explain it so, as I said, you can resolve this comment as nothing happened :)